src/Helper/HeaderHelper.php

```diff
@@ -6,8 +6,37 @@
 
 final class HeaderHelper
 {
+    /**
+     * @link https://www.rfc-editor.org/rfc/rfc2616.html#section-2.2
+     * token  = 1*<any CHAR except CTLs or separators>
+     */
+    private const PATTERN_TOKEN = '(?:(?:[^()<>@,;:\\"\/[\\]?={} \t\x7f]|[\x00-\x1f])+)';
+
+    /**
+     * @link https://www.rfc-editor.org/rfc/rfc2616.html#section-3.6
+     * attribute = token
+     */
+    private const PATTERN_ATTRIBUTE = self::PATTERN_TOKEN;
+
+    /**
+     * @link https://www.rfc-editor.org/rfc/rfc2616.html#section-2.2
+     * quoted-string  = ( <"> *(qdtext | quoted-pair ) <"> )
+     * qdtext         = <any TEXT except <">>
+     * quoted-pair    = "\" CHAR
+     */
+    private const PATTERN_QUOTED_STRING = '(?:"(?:(?:\\\\.)+|[^\\"]+)*")';
+
+    /**
+     * @link https://www.rfc-editor.org/rfc/rfc2616.html#section-3.6
+     * value = token | quoted-string
+     */
+    private const PATTERN_VALUE = '(?:' . self::PATTERN_QUOTED_STRING . '|' . self::PATTERN_TOKEN . ')';
+
     /**
      * Explode header value to value and parameters (eg. text/html;q=2;version=6)
+     *
+     * @link https://www.rfc-editor.org/rfc/rfc2616.html#section-3.6
+     * transfer-extension      = token *( ";" parameter )
      * @param string $headerValue
      * @return array first element is the value, and key-value are the parameters
```

Comment:
What do you think about those cases when the values of the header are specified by a comma separated list?
Should this function return a list of arrays?
`Accept: audio/*; q=0.2, audio/basic`
```
[
  ['audio/*', 'q' => '0.2'],
  ['audio/basic'],
]
```
If so, can it be better to pass a `Request` object instead `$headerValue` parameter here with the name of the header, or an array of strings received with `$request->getHeader($headerName)`?

If we get only one value, we need to write that the list is not processed and we get only the first value